paper.tex

```diff
@@ -207,34 +207,86 @@ \section*{Demes}
+the duration of their existences, and population sizes are given as the number
+of individuals. A population must exist for some duration of time greater than
+zero, so that its start time must be larger than its end time. It may have one
+or more ancestors, which are other populations that exist at the population's
+start time. If one ancestor is specified, the first generation is constructed
+by randomly sampling parents from the ancestral population to contribute to
+offspring in the newly generated population. If more than one ancestor is
+specified, the proportions of ancestry from each contributing population must
+be provided, and those proportions must sum to one. In this case, parents are
+chosen randomly from each ancestral population with probability given by those
+proportions. If no ancestors are specified, the population is assumed to have
+start time equal to infinity, as individuals are not assumed to spontaneously
+spring into existence.
+
+Within populations, population sizes and mating systems are specified within
+epochs. Epochs are non-overlapping intervals that cover the entire existence
+interval of the population. Each epoch specifies the population size over that
+interval, which can be a constant value or function defined by start and end
+sizes that must remain positive.  If an epoch has a start time of infinity, the
+population size for that epoch must be constant. Epochs can also specify
+parameters for nonrandom mating, such as selfing or cloning rates, which give
+the probability that offspring are generated from one generation to the next by
+self-fertilization or cloning of an indivual \textbf{need to precisely specify
+once this is determined}.
+
+Finally, individuals in a population may have parents from a different
+population through migrations. These can be defined as continuous migration
+rates over intervals that populations overlap in time or through instantaneous
+(or pulse) migration events at a given time. Continuous migration rates are
+defined as the probability that parents in the ``destination`` population are
+chosen from the ``source`` population.  Migration rates are thus per-generation
+and must be less than one. Furthermore, if more than one source population have
+continuous migration into the same destination population, the sum of those
+migration rates must also be less than one, as rates define probabilities. The
+probability that parents come from the same population is just one minus the
+some of incoming migration rates. On the other hand, pulse migration events
+specify the instantaeous replacement of a given fraction of individuals in a
+destination population by individuals with parents from a source population.
+The fraction must be between zero and one, and if more than one source
+population has a pulse migration event to the same destination population and
+the same time, those replacement events are applied sequentially in the order
+that they are specified in the model.
```

Comment:
This is not quite sufficient, if one considers two pulses like A->B, B->C. The order here matters, but the pulse destination is different in each case.
```suggestion
The fraction must be between zero and one, and if more than one pulse is defined
for the same time, those replacement events are applied sequentially in the order
that they are specified in the model.
```